app/src/test/java/protect/card_locker/DatabaseTest.java

```diff
@@ -317,44 +317,44 @@ public void updateGroup() {
         assertEquals(1, DBHelper.getGroupCount(mDatabase));
 
         // Add card to group
-        Group group = DBHelper.getGroup(mDatabase, "group one");
+        Group group = DBHelper.getGroupByName(mDatabase, "group one");
         List<Group> groupList1 = new ArrayList<>();
         groupList1.add(group);
         DBHelper.setLoyaltyCardGroups(mDatabase, 1, groupList1);
 
         // Ensure the card has one group and the group has one card
         List<Group> cardGroups = DBHelper.getLoyaltyCardGroups(mDatabase, (int) id);
         assertEquals(1, cardGroups.size());
-        assertEquals("group one", cardGroups.get(0)._id);
-        assertEquals(1, DBHelper.getGroupCardCount(mDatabase, "group one"));
+        assertEquals("group one", cardGroups.get(0).name);
+        assertEquals(1, DBHelper.getGroupCardCount(mDatabase, group._id));
 
         // Rename group
-        result = DBHelper.updateGroup(mDatabase, "group one", "group one renamed");
+        result = DBHelper.updateGroup(mDatabase, group._id, "group one renamed");
         assertTrue(result);
         assertEquals(1, DBHelper.getGroupCount(mDatabase));
 
         // Group one no longer exists
-        group = DBHelper.getGroup(mDatabase,"group one");
+        group = DBHelper.getGroupByName(mDatabase,"group one");
         assertNull(group);
 
         // But group one renamed does
-        Group group2 = DBHelper.getGroup(mDatabase, "group one renamed");
+        Group group2 = DBHelper.getGroupByName(mDatabase, "group one renamed");
```

Comment:
We should also test if we can get a group by ID, as that's the actual canonical thing.